File: CRM_Admin/app/src/main/java/com/irg/crm_admin/activity/SplashActivity.java
package com.irg.crm_admin.activity;

import android.content.Context;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ProgressBar;

import androidx.appcompat.app.AppCompatActivity;

import com.irg.crm_admin.R;
import com.irg.crm_admin.common.OprActivity;
import com.irg.crm_admin.common.SharedPref;

public class SplashActivity extends AppCompatActivity {

    Context context;
    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_splash);

        initView();
    }

    private void initView() {
        context = SplashActivity.this;
        //progressBar = findViewById(R.id.progressBar);
        openDasboard();
    }

    private void openDasboard() {

      /*  new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                final Intent mainIntent = new Intent(SplashActivity.this, MainActivity.class);
                SplashActivity.this.startActivity(mainIntent);
                SplashActivity.this.finish();
            }
        }, 1000);*/

        if (!TextUtils.isEmpty(SharedPref.getMobileNo(context)) && !TextUtils.isEmpty(SharedPref.getPassword(context))) {
            LoginActivity.attemptLogin(context, SharedPref.getMobileNo(context), SharedPref.getPassword(context));
        } else {
            OprActivity.finishAllOpenNewActivity(context, new LoginActivity());
        }
    }
}

File: SWEEP/app/src/main/java/com/csows/sweep/activities/SplashActivity.kt
package com.csows.sweep.activities

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.text.TextUtils
import android.view.View
import android.view.Window
import android.view.WindowManager
import com.csows.sweep.R
import com.csows.sweep.utils.ConnectionDetector
import com.csows.sweep.utils.UserPreferences
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : AppCompatActivity() {

    private lateinit var mCtx:Context
    lateinit var userPreferences: UserPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        this.window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )
        setContentView(R.layout.activity_splash)

        initView()

    }

    private fun initView() {
        this.also { mCtx = it }
        userPreferences = UserPreferences(mCtx)
        progressbar.visibility = View.VISIBLE
        if (ConnectionDetector.isConnectedToInernet(mCtx)) {

            val loginActivity = LoginActivity()
            if (userPreferences.getLogin()) {
                if (!TextUtils.isEmpty(userPreferences.getOfficerMobile())) {
                    loginActivity.getLogin(userPreferences.getOfficerMobile()!!, false, mCtx)
                } else {
                    startActivity(Intent(mCtx, LoginActivity::class.java))
                    finish()
                }
            }else{
                startActivity(Intent(mCtx, LoginActivity::class.java))
                finish()
            }

        }else{
            showSnakbar()
        }

    }


    private fun showSnakbar() {
        progressbar.visibility = View.GONE
        Snackbar.make(findViewById(R.id.parent),
            resources.getString(R.string.no_internet),
            Snackbar.LENGTH_INDEFINITE
        )
            .setAction("Retry") {
                initView()
            }
            .setActionTextColor(resources.getColor(android.R.color.holo_blue_dark))
            .show()
    }
}